### PR TITLE
Use the Keras set_random_seed in tests

### PR DESCRIPTION
**Pull request by Rocketknight1**
Some tests load checkpoints where weights are missing (e.g. the `pooler` weights for `ViT`). To get reproducible results, we need to ensure that seeds are set consistently. This PR uses `keras.set_random_seed()` instead of `tf.set_seed()`, which fixes some issues the CI was seeing where models mismatched because of differing weight inits.

### CODE REVIEW

File: tests/pipelines/test_pipelines_common.py [hunk 1]
```diff
@@ -541,19 +541,18 @@ def test_load_default_pipelines_pt(self):
     @slow
     @require_tf
     def test_load_default_pipelines_tf(self):
-        import tensorflow as tf
-
+        from transformers.modeling_tf_utils import keras
         from transformers.pipelines import SUPPORTED_TASKS
 
-        set_seed_fn = lambda: tf.random.set_seed(0)  # noqa: E731
+        set_seed_fn = lambda: keras.utils.set_random_seed(0)  # noqa: E731
```
Comment (amyeroberts) [inline tests/pipelines/test_pipelines_common.py:547]: So tf.random.set_seed doesn't actually set the seed for the TF environment? 

Does keras.utils.set_random_seed control just keras objects, or the whole TF env too?

**Review by ydshieh — APPROVED** [review]
thanks @Rocketknight1 

(is this a API changes causing the problem?)

File: tests/pipelines/test_pipelines_common.py (hunk 1 above)
Comment (Rocketknight1) [reply tests/pipelines/test_pipelines_common.py:547]: As far as I can tell, `keras.utils.set_random_seed()` basically sets a bunch of seeds, which includes `tf.random.set_seed()`, but also some others. It's the [currently recommended approach for reproducible initialization](https://keras.io/examples/keras_recipes/reproducibility_recipes/) in the Keras docs.

Comment (ydshieh) [reply tests/pipelines/test_pipelines_common.py:547]: Hope there is no other tests having the same init issue but no failure by luck 🤞 .
(otherwise we might get different results for them after this PR 😅 )

Comment (amyeroberts) [reply tests/pipelines/test_pipelines_common.py:547]: I'm not surprised it's the recommended one in the keras docs ;) 

OK, well I just we just shrug and put it in the list of other TF quirks

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!
